test/api/importcsv/api_test.go

```diff
@@ -0,0 +1,166 @@
+			expectedApps := r.ExpectedApplications
+			if len(gotApps) != len(expectedApps) {
+				t.Errorf("Mismatch in number of imported Applications: Expected %d, Actual %d", len(expectedApps), len(gotApps))
+			} else {
+				for i, importedApp := range gotApps {
+					if expectedApps[i].Name != importedApp.Name {
+						t.Errorf("Mismatch in name of imported Application: Expected %s, Actual %s", expectedApps[i].Name, importedApp.Name)
+					}
+					if expectedApps[i].Description != importedApp.Description {
+						t.Errorf("Mismatch in description of imported Application: Expected %s, Actual %s", expectedApps[i].Description, importedApp.Description)
+					}
+					if expectedApps[i].Repository.Kind != importedApp.Repository.Kind {
+						t.Errorf("Mismatch in repository's kind ofimported Application: Expected %s, Actual %s", expectedApps[i].Repository.Kind, importedApp.Repository.Kind)
+					}
+					if expectedApps[i].Repository.URL != importedApp.Repository.URL {
+						t.Errorf("Mismatch in repository's url of imported Application: Expected %s, Actual %s", expectedApps[i].Repository.URL, importedApp.Repository.URL)
+					}
+					if expectedApps[i].Binary != importedApp.Binary {
+						t.Errorf("Mismatch in binary of imported Application: Expected %s, Actual %s", expectedApps[i].Binary, importedApp.Binary)
+					}
+					for j, tag := range expectedApps[i].Tags {
+						if tag.Name != importedApp.Tags[j].Name {
+							t.Errorf("Mismatch in tag name of imported Application: Expected %s, Actual %s", tag.Name, importedApp.Tags[j].Name)
+						}
+					}
+					if expectedApps[i].BusinessService.Name != importedApp.BusinessService.Name {
+						t.Errorf("Mismatch in name of the BusinessService of imported Application: Expected %s, Actual %s", expectedApps[i].BusinessService.Name, importedApp.BusinessService.Name)
+					}
+				}
+			}
+
+			// Check list of Dependencies.
+			gotDeps, _ := Dependency.List()
+			expectedDeps := r.ExpectedDependencies
+			if len(gotDeps) != len(expectedDeps) {
+				t.Errorf("Mismatch in number of imported Dependencies: Expected %d, Actual %d", len(expectedDeps), len(gotDeps))
+			} else {
+				for i, importedDep := range gotDeps {
+					expectedDep := expectedDeps[i].To.Name
+					if importedDep.To.Name != expectedDep {
+						t.Errorf("Mismatch in imported Dependency: Expected %s, Actual %s", expectedDep, importedDep.To.Name)
+					}
+				}
+			}
+
+			// fetch id of CSV file and convert it into required formats
+			var inputID = strconv.FormatUint(uint64(inputData.ID), 10) // to be used for API compatibility
+
+			var outputImportSummaries []api.ImportSummary
+			outputMatchingSummary := api.ImportSummary{}
+			assert.Should(t, Client.Get(api.SummariesRoot, &outputImportSummaries))
+			for _, imp := range outputImportSummaries {
+				if uint(imp.ID) == inputData.ID {
+					outputMatchingSummary = imp
+				}
+			}
+			assert.FlatEqual(len(expectedApps)+len(expectedDeps), outputMatchingSummary.ValidCount)
```

Comment:
Already commented the same thing some lines above in this file. ```assert.FlatEqual``` returns true if arguments are equal, false if different - that doesn't fail the test. Use it in condition like:
```if !assert.FlatEqual(...) { t.Error(error_results_are_different) }```

The advantage of assert.FlatEqual is that it checks multiple attributes on struct types variables. For basic types line string or int, it might be shorter use just == operator in the condition.